### ISSUE TITLE
学習率スケジューラが毎バッチで呼ばれている

### ISSUE THREAD

**Issue by Ktomoya912** [Ktomoya912/contrast_alphazero#8]
## 概要

`main.py` で `scheduler.step()` が各バッチ更新後に呼ばれていますが、`StepLR` スケジューラはエポック単位での呼び出しを想定しています。

## 問題箇所

`main.py:301`

```python
optimizer.step()
scheduler.step()  # ← 毎バッチで呼ばれている
```

## 影響

`LR_STEP_SIZE = 50,000` の設定は、50,000エポックごとに学習率を減衰させることを意図していると思われますが、実際には50,000バッチ（= 50,000回のミニバッチ更新）で減衰してしまいます。

これにより、学習率が意図した速度よりもはるかに速く減少し、学習が早期に停滞する可能性があります。

## 提案される修正

1. セルフプレイゲーム終了ごとに `scheduler.step()` を呼ぶ
2. または、`LR_STEP_SIZE` を意図したバッチ数に調整する
3. または、バッチ単位のスケジューラ（例：`OneCycleLR`）に変更する

**Comment by Ktomoya912** [comment]
確認の結果、現在の実装は意図通りでした。`scheduler.step()`は各学習ステップ（セルフプレイ1ゲーム完了後）に呼ばれ、`LR_STEP_SIZE=50,000`ステップでLRが半減する設計です。AlphaZero形式の継続的学習ではエポック概念がないため、この動作は適切です。
